src/cljam/io/bam/writer.clj

```diff
@@ -53,14 +78,46 @@
       (lsb/write-int w (:len ref)))))
 
 (defn write-alignments* [^BAMWriter wtr alns header]
-  (let [w (.writer wtr)
+  (let [dw (.data-writer wtr)
+        w ^BGZFOutputStream (.writer wtr)
         refs (refs/make-refs header)]
-    (doseq [a alns]
-      (lsb/write-int w (encoder/get-block-size a))
-      (encoder/encode-alignment w a refs))))
+    (with-open [baos (java.io.ByteArrayOutputStream. 4096)
+                dos (java.io.DataOutputStream. baos)]
```

Comment:
Add `ByteArrayOutputStream` to the `import` clause of `ns`

```suggestion
    (with-open [baos (ByteArrayOutputStream. 4096)
                dos (DataOutputStream. baos)]
```

Reply:
Fixed.